src/repositories/network/network_redis.rs

```diff
@@ -0,0 +1,982 @@
+        })?;
+
+        debug!("Successfully updated indexes for network {}", network.id);
+        Ok(())
+    }
+
+    /// Remove all indexes for a network
+    async fn remove_all_indexes(&self, network: &NetworkRepoModel) -> Result<(), RepositoryError> {
+        let mut conn = self.client.as_ref().clone();
+        let mut pipe = redis::pipe();
+        pipe.atomic();
+
+        debug!("Removing all indexes for network {}", network.id);
+
+        // Remove name index
+        let name_key = self.network_name_index_key(&network.network_type, &network.name);
+        pipe.del(&name_key);
+
+        // Remove chain ID index if applicable
+        if let Some(chain_id) = self.extract_chain_id(network) {
+            let chain_id_key = self.network_chain_id_index_key(&network.network_type, chain_id);
+            pipe.del(&chain_id_key);
+            debug!(
+                "Removing chain ID index for network {} with chain_id {}",
+                network.id, chain_id
+            );
+        }
+
+        pipe.exec_async(&mut conn).await.map_err(|e| {
+            error!("Index removal failed for network {}: {}", network.id, e);
+            self.map_redis_error(e, &format!("remove_indexes_for_network_{}", network.id))
+        })?;
+
+        debug!(
+            "Successfully removed all indexes for network {}",
+            network.id
+        );
+        Ok(())
+    }
+
+    /// Batch fetch networks by IDs
+    async fn get_networks_by_ids(
+        &self,
+        ids: &[String],
+    ) -> Result<Vec<NetworkRepoModel>, RepositoryError> {
+        if ids.is_empty() {
+            debug!("No network IDs provided for batch fetch");
+            return Ok(vec![]);
+        }
+
+        let mut conn = self.client.as_ref().clone();
+        let keys: Vec<String> = ids.iter().map(|id| self.network_key(id)).collect();
+
+        debug!("Batch fetching {} networks", ids.len());
+
+        let values: Vec<Option<String>> = conn
+            .mget(&keys)
+            .await
+            .map_err(|e| self.map_redis_error(e, "batch_fetch_networks"))?;
+
+        let mut networks = Vec::new();
+        let mut failed_count = 0;
+
+        for (i, value) in values.into_iter().enumerate() {
+            match value {
+                Some(json) => {
+                    match self.deserialize_entity::<NetworkRepoModel>(&json, &ids[i], "network") {
+                        Ok(network) => networks.push(network),
+                        Err(e) => {
+                            failed_count += 1;
+                            error!("Failed to deserialize network {}: {}", ids[i], e);
+                        }
+                    }
+                }
+                None => {
+                    warn!("Network {} not found in batch fetch", ids[i]);
+                }
+            }
+        }
+
+        if failed_count > 0 {
+            warn!(
+                "Failed to deserialize {} out of {} networks in batch",
```

Comment:
Maybe making this error more transparent by passing the deserealization errors as part of the returned value, this way the caller can figure out if the network is not present because failed to deserealize or because it's missing

Ok(NetworksIdsResult { 
  networks: ...,
  errors: Option<...>
})

Reply:
Improved by returning failed ids

Thanks for suggesting